led_strip/include/led_strip_spi.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "driver/gpio.h"
+#include "esp_rom_gpio.h"
+#include "driver/spi_master.h"
+#include "driver/spi_common.h"
+#include "hal/spi_types.h"
+#include "led_strip_types.h"
+#include "soc/gpio_sig_map.h"
+#include "soc/spi_periph.h"
```

Comment:
Don't expose private header files in a public one.

```suggestion
#include "driver/spi_master.h"
#include "led_strip_types.h"
```

Reply:
you can include the private header files in the `.c` file